lib/cargo.js

```diff
@@ -372,17 +372,19 @@ export function provideBuilder() {
 
       // This function is called before every build. It finds the closest
       // Cargo.toml file in the path and uses its directory as working.
-      const preBuildFunction = multiCrateProjects && function () {
-        const editor = atom.workspace.getActiveTextEditor();
-        this.cwd = undefined;
-        if (editor && editor.getPath()) {
-          const wdInfo = findCargoProjectDir(editor.getPath());
-          if (wdInfo) {
-            if (!wdInfo.root) {
-              const p = path.parse(wdInfo.dir);
-              atom.notifications.addInfo('Building ' + p.base + '...');
+      const preBuildFunction = function () {
+        if (multiCrateProjects) {
```

Comment:
Another possible solution would be to leave this function intact and check `buildWorkDir` wherever we use it. There's only one such place now, so it's not a big deal. I could switch to this approach if you don't feel confident with using this hook by default. But for me, it seems rather safe since most of the logic is switched off by `multiCrateProjects` setting.